tests/snapshot.rs

```diff
@@ -38,49 +89,181 @@ struct Report {
+                    VadTransition::SpeechStart { timestamp_ms: ts_2 },
+                ) => {
+                    if ts_1 != ts_2 {
+                        return false;
+                    }
+                }
+                (
+                    VadTransition::SpeechEnd {
+                        start_timestamp_ms: start_1,
+                        end_timestamp_ms: end_1,
+                        ..
+                    },
+                    VadTransition::SpeechEnd {
+                        start_timestamp_ms: start_2,
+                        end_timestamp_ms: end_2,
+                        ..
+                    },
+                ) => {
+                    if start_1 != start_2 {
+                        return false;
+                    } else if end_1 != end_2
+                        && (*end_1 as isize - *end_2 as isize).abs() as usize
+                            != allowed_post_pad_diff
+                    {
+                        return false;
+                    }
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        true
+    }
+
+    fn eq_current_session_samples(
+        &self,
+        other: &Self,
+        _self_config: &VadConfig,
+        _other_config: &VadConfig,
+    ) -> bool {
+        self.current_session_samples == other.current_session_samples
+    }
+
+    fn eq_current_speech_samples(
+        &self,
+        other: &Self,
+        self_config: &VadConfig,
+        other_config: &VadConfig,
+    ) -> bool {
+        // Some differences are introduced by modified config and they are expected.
+        let allowed_post_pad_diff = (self_config.post_speech_pad.as_millis() as isize
+            - other_config.post_speech_pad.as_millis() as isize)
+            .abs() as usize
+            * self_config.sample_rate
+            / 1000;
+
+        for (baseline, current) in self
+            .current_speech_samples
+            .iter()
+            .zip(other.current_speech_samples.iter())
+        {
+            if baseline != current
+                && (*baseline as isize - *current as isize).abs() as usize != allowed_post_pad_diff
+            {
+                return false;
+            }
+        }
+        true
+    }
+
+    fn eq_current_silence_samples(
+        &self,
+        other: &Self,
+        _self_config: &VadConfig,
+        _other_config: &VadConfig,
+    ) -> bool {
+        self.current_silence_samples == other.current_silence_samples
+    }
+
+    fn eq_likelihoods(
+        &self,
+        other: &Self,
+        _self_config: &VadConfig,
+        _other_config: &VadConfig,
+    ) -> bool {
+        self.likelihoods == other.likelihoods
+    }
+}
+
 #[test]
 #[traced_test]
 fn chunk_50_default_params_16k() {
-    run_snapshot_test(50, VadConfig::default(), "default");
+    let mut config = VadConfig::default();
+
+    for post_speech_pad in [0, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600] {
```

Comment:
I think this just complicates the snapshot tests a bit too much. Plus the `default_params` part of the name doesn't mean much when we run with 13 different configs 12 of which have non-default params.

I think what would be better is just doing a non-snapshot post-speech-pad test where you load one of the files with multiple speech segments. Run it with no post-speech-pad  and with some padding value above 0 and less than the redemption time. Then make sure the `SpeechEnd` timestamps have the padding added on and there's more samples than in the unpadded version.

There's some smaller simpler silero tests like this at the bottom of `src/lib.rs`